common/cluster/archival_status.go

```diff
@@ -31,19 +31,21 @@ type (
 	// ArchivalStatus represents the archival status of the cluster
 	ArchivalStatus int
 
-	// ArchivalConfig is an immutable representation of the current archival configuration of the cluster
+	// ArchivalConfig is an immutable representation of the archival configuration of the cluster
+	// This config is determined at cluster startup time
```

Comment:
Can we include a comment in server.go: 

"Dynamic archival config is accessed once on cluster startup than never accessed again. This is done so as to keep archival status and and the initialization of archiver.Provider in sync."

Reply:
This comment should be added in addition to the comment you added here. I suggest adding this comment because it is pretty confusing why this archival config requires a restart.

Reply:
Can we also add a TODO in server go that says "TODO: Once archival pause is implemented archival config can be made truly dynamic."